LutronLuxmeterIntegration/src/reader.ts

```diff
@@ -0,0 +1,69 @@
+import * as fs from 'fs/promises';
+import { SerialPort } from 'serialport';
+
+export type TResult = { value: number; unit: string };
+
+const bufferSize = 16;
+
+export class LuxMeterReader {
+    private path = '';
+    private validResponse(buf: Buffer): boolean {
+        const STX = 2;
+        const LF = 10;
+        const zeroCode = '0'.charCodeAt(0);
+        const nineCode = '9'.charCodeAt(0);
+
+        if (buf.length !== bufferSize || buf[0] !== STX || buf.at(-1) !== LF) {
+            return false;
+        }
+        for (let f = 1; f < bufferSize - 1; f += 1) {
+            if (zeroCode > buf[f] || buf[f] > nineCode) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+    private async findLuxMeter(): Promise<void> {
+        const list = await SerialPort.list();
+
+        for (const port of list) {
+            const buf = await this.read(port.path);
+
+            if (this.validResponse(buf)) {
+                this.path = port.path;
+                return;
+            }
+        }
+
+        throw new Error('No Lux meter found.');
+    }
+
+    private constructor() {}
```

Comment:
konstruktor vetsinou mame jako prvni metodu..jasne je privatni, ale i tak bych to dal nahori i s tim connect..

Reply:
Pořadí funkcí mi dá vždycky zabrat.